ragaai_catalyst/synthetic_data_generation.py

```
@@ -476,6 +528,228 @@ def get_supported_providers(self):
+**Final Notes**:  
+- Focus on both originality and practical relevance.
+- Avoid repetitiveness in the examples.
+'''
+        return prompt
+    
+    def _generate_examples_iter(
+            self, 
+            user_instruction: str, 
+            user_examples: Optional[List[str] | str] = None, 
+            user_context: Optional[str] = None, 
+            relevant_examples: List[str]=[], irrelevant_examples: List[str]=[], 
```

Ruff (0.8.2)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)